data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSinkConfig.java

```diff
@@ -0,0 +1,194 @@
+package org.opensearch.dataprepper.plugins.sink.opensearch;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import lombok.Getter;
+import org.apache.commons.lang3.EnumUtils;
+import org.opensearch.dataprepper.model.opensearch.OpenSearchBulkActions;
+import org.opensearch.dataprepper.plugins.sink.opensearch.configuration.ActionConfiguration;
+import org.opensearch.dataprepper.plugins.sink.opensearch.configuration.AwsAuthenticationConfiguration;
+import org.opensearch.dataprepper.plugins.sink.opensearch.configuration.DlqConfiguration;
+import org.opensearch.dataprepper.plugins.sink.opensearch.index.TemplateType;
+
+import java.util.List;
+import java.util.Objects;
+
+public class OpenSearchSinkConfig {
```

Comment:
I think we are missing some fields. Some of these are deprecated, but they need to remain until Data Prepper 3.0.

* `aws_sigv4`
* `aws_region`
* `aws_sts_role_arn`
* `aws_sts_external_id`
* `aws_sts_header_overrides`
* `document_id_field`
* `routing_field`

There may be some others as well.